src/cbica_toolkit/src/cbicaITKUtilities.h

```diff
@@ -557,13 +581,36 @@ namespace cbica
         {
           if (imageDirs1[d][i] != imageDirs2[d][i])
           {
-            auto percentageDifference = std::abs(imageDirs1[d][i] - imageDirs2[d][i]) * 100 / imageDirs1[d][i];
+            auto percentageDifference = std::abs(imageDirs1[d][i] - imageDirs2[d][i]);
+            // perform an absolute check when either of the direction cosine location is zero
+            if ((imageDirs1[d][i] == 0) || (imageDirs2[d][i] == 0))
+            {
+              auto threshold = 10e-5; // chose as a very small number, which is greater than std::numeric_traits<float>::epsilon() for comparison purposes
+              if (percentageDifference < threshold)
+              {
+                std::cerr << "Ignoring absolute difference of " << percentageDifference << " at location '[" << d << "," << i << "]' of direction matrix.\n";
+                percentageDifference = 0;
+              }
+              else
+              {
+                // otherwise, calculate the percentage based on the non-zero cosine
+                if (imageDirs1[d][i] == 0)
+                {
+                  percentageDifference /= imageDirs2[d][i];
+                }
+                else if (imageDirs2[d][i] == 0)
+                {
+                  percentageDifference /= imageDirs1[d][i];
+                }
+              }
+            }
+            percentageDifference *= 100;
```

Comment:
Possible code duplication. Better to put into a function.

Reply:
Done, thanks for pointing it out.